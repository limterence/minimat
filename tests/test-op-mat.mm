/* test-op-mat: matrix operators */
int main() {
  matrix u;
  matrix v;
  matrix w;
  matrix y;
  matrix x;
  u = [1.0, 2.0, 3.0, 4.0; 11.0, 12.0, 13.0, 14.0;];
  v = u;
  v[0,0] = [1000.0;];
  printmat(u);
  printmat(v);
  printmat(u + v);             /* arithmetic */
  printmat(v + u);
  printmat(u - v);
  printmat(v - u);
  printmat([1.0, 2.0; 3.0, 4.0;] ^ 3);
  printmat(-v);                /* unary */
  printmat(v');
  printmat(v + [1000.0;]);     /* arithmetic with scalar matrix*/
  printmat([2000.0;] - v);
  printmat(u .* u);            /* elemental */
  printmat(v ./ u);
  printmat(v .% u);
  printmat(v .^ [3.0;]);
  printseq(u < v);             /* relational */
  printseq(u <= v);
  printseq(u > v);
  printseq(u >= v);
  printseq(u == v);
  printseq(u != v);

  printmat(u * u');                      /* matrix multiplication */
  y = [2.0; 0.5; 1.5; 5.0; 7.0; 7.0;];   /* matrix regression */
  x = [1.0,2.0; 2.0,2.0; 3.0,3.0; 4.0,3.0; 5.0,5.0; 6.0,6.0;];
  x = [new matrix(rows(x), 1) + [1.0;], x;];
  printmat(y/x);
  printmat(y%x);
}
