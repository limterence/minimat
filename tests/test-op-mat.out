[2 x 4 float]
  1.00   2.00   3.00   4.00 
 11.00  12.00  13.00  14.00 
[2 x 4 float]
1000.00   2.00   3.00   4.00 
 11.00  12.00  13.00  14.00 
[2 x 4 float]
1001.00   4.00   6.00   8.00 
 22.00  24.00  26.00  28.00 
[2 x 4 float]
1001.00   4.00   6.00   8.00 
 22.00  24.00  26.00  28.00 
[2 x 4 float]
-999.00   0.00   0.00   0.00 
  0.00   0.00   0.00   0.00 
[2 x 4 float]
999.00   0.00   0.00   0.00 
  0.00   0.00   0.00   0.00 
[2 x 2 float]
 37.00  54.00 
 81.00 118.00 
[2 x 4 float]
-1000.00  -2.00  -3.00  -4.00 
-11.00 -12.00 -13.00 -14.00 
[4 x 2 float]
1000.00  11.00 
  2.00  12.00 
  3.00  13.00 
  4.00  14.00 
[2 x 4 float]
2000.00 1002.00 1003.00 1004.00 
1011.00 1012.00 1013.00 1014.00 
[2 x 4 float]
1000.00 1998.00 1997.00 1996.00 
1989.00 1988.00 1987.00 1986.00 
[2 x 4 float]
  1.00   4.00   9.00  16.00 
121.00 144.00 169.00 196.00 
[2 x 4 float]
1000.00   1.00   1.00   1.00 
  1.00   1.00   1.00   1.00 
[2 x 4 float]
  0.00   0.00   0.00   0.00 
  0.00   0.00   0.00   0.00 
[2 x 4 float]
1000000000.00   8.00  27.00  64.00 
1331.00 1728.00 2197.00 2744.00 
[1 int]
0 
[8 int]
0 1 2 3 4 5 6 7 
[0 int]
[7 int]
1 2 3 4 5 6 7 
[7 int]
1 2 3 4 5 6 7 
[1 int]
0 
[2 x 2 float]
 30.00 130.00 
130.00 630.00 
[3 x 1 float]
 -1.42 
  0.75 
  0.75 
[6 x 1 float]
  1.17 
 -1.08 
 -1.58 
  1.17 
  0.92 
 -0.58 
